CTLD.lua

```diff
@@ -5991,9 +5993,8 @@ function ctld.createSmokeMarker(_enemyUnit, _colour)
     --recreate in 5 mins
     ctld.jtacSmokeMarks[_enemyUnit:getName()] = timer.getTime() + 300.0
 
-    -- move smoke 2 meters above target for ease
     local _enemyPoint = _enemyUnit:getPoint()
-    trigger.action.smoke({ x = _enemyPoint.x + ctld.JTAC_smokeOffset_x, y = _enemyPoint.y + ctld.JTAC_smokeOffset_y, z = _enemyPoint.z + ctld.JTAC_smokeOffset_z }, _colour)
+    trigger.action.smoke({ x = _enemyPoint.x + math.random(0, ctld.JTAC_smokeMarginOfError) + ctld.JTAC_smokeOffset_x, y = _enemyPoint.y + ctld.JTAC_smokeOffset_y, z = _enemyPoint.z + math.random(0, ctld.JTAC_smokeMarginOfError) + ctld.JTAC_smokeOffset_z }, _colour)
```

Comment:
I would propose making the math.random positive and negative so the smoke marker is not always NE of the target:

    trigger.action.smoke({ x = _enemyPoint.x + math.random(-1 * ctld.JTAC_smokeMarginOfError, ctld.JTAC_smokeMarginOfError) + ctld.JTAC_smokeOffset_x,
                           y = _enemyPoint.y + ctld.JTAC_smokeOffset_y,
                           z = _enemyPoint.z + math.random(-1 * ctld.JTAC_smokeMarginOfError, ctld.JTAC_smokeMarginOfError) + ctld.JTAC_smokeOffset_z },
                           _colour)

Reply:
good point, thank you.